File: src/backend/app/Models/Booking.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Booking extends Model
{
    const BOOKING_STATUS_CREATED    = 'Created';
    const BOOKING_STATUS_CONFIRMED  = 'Confirmed';
    const BOOKING_STATUS_CANCELLED  = 'Cancelled';
    const BOOKING_STATUS_FINISHED   = 'Finished';

    protected $table = 'booking';

    protected $primaryKey = 'bookingId';

    public $timestamps = false;

    protected $fillable = [
        'lodgingId',
        'customerId',
        'status',
        'startDate',
        'endDate',
        'fees'
    ];

    public function lodging()
    {
        return $this->belongsTo(Lodging::class, 'lodgingId');
    }

    public function customer()
    {
        return $this->belongsTo(Person::class, 'customerPersonId');
    }

    public function payment()
    {
        return $this->hasOne(Payment::class, 'bookingId');
    }
}

File: src/backend/src/api/Models/Booking.cs
namespace Restify.API.Models;

public class Booking
{
	public int				Id { get; set; }
	public int				CustomerId { get; set; }
	public int				LodgingId { get; set; }

	public List<RoomBooking>	RoomBookings { get; }
	public Payment?				Payment { get; set; }
	public Person				Customer { get; }
	public Lodging				Lodging { get; }

	public Booking() {}
	
	public Booking(List<RoomBooking> roomBookings)
	{
		RoomBookings = roomBookings;
	}
}

public enum BookingStatus
{
	Created,
	Confirmed,
	Cancelled,
	Finished
}
